src/api/instance/localInstanceImpl.js

```diff
@@ -0,0 +1,184 @@
+// @flow
+
+import {Instance} from "./instance";
+import {type CredrankInput} from "../credrank";
+import {
+  type WeightsT,
+  parser as weightsParser,
+  empty as emptyWeights,
+} from "../../core/weights";
+import fs from "fs-extra";
+import {join as pathJoin} from "path";
+import stringify from "json-stable-stringify";
+import {
+  type WeightedGraph,
+  fromJSON as weightedGraphFromJSON,
+} from "../../core/weightedGraph";
+import {
+  loadJson,
+  mkdirx,
+  loadFileWithDefault,
+  loadJsonWithDefault,
+} from "../../util/disk";
+import {parser as configParser, type InstanceConfig} from "../instanceConfig";
+import {Ledger} from "../../core/ledger/ledger";
+import {
+  parser as dependenciesParser,
+  type DependenciesConfig,
+} from "../dependenciesConfig";
+import {type Budget} from "../../core/mintBudget";
+import {parser as pluginBudgetParser} from "../pluginBudgetConfig";
+import {
+  CredGraph,
+  parser as credGraphParser,
+} from "../../core/credrank/credGraph";
+
+const DEPENDENCIES_PATH = ["config", "dependencies.json"];
+const WEIGHT_OVERRIDES_PATH = ["config", "weights.json"];
+const BUDGET_PATH = ["config", "pluginBudgets.json"];
+const INSTANCE_CONFIG_PATH = ["sourcecred.json"];
+const CREDGRAPH_PATH = ["output", "credGraph.json"];
+const GRAPHS_DIRECTORY = ["output", "graphs"];
+const GRAPHS_PATH = ["graph.json"];
+const LEDGER_PATH = ["data", "ledger.json"];
+
+/**
+This is an Instance implementation that reads and writes using relative paths
+on the local disk.
+ */
+export class LocalInstance implements Instance {
```

Comment:
Not attached. I actually generally see the class name include the suffix too. Or sometimes instead of the "Impl" suffix, the interface gets an "I" suffix. Makes it a little easier to delineate, but I don't really care.

Reply:
My thoughts here:

  - Interfaces are the most important, so they should always get the
    good names. Thus, `List`, not `IList`. So, I approve your choice of
    interface name `Instance`.

  - For concrete implementations, the “distinguishing characteristic
    followed by interface name” pattern works well: `ArrayList`,
    `BrowserLocalStore`, etc. So, I approve your choice of class name
    `LocalInstance`.

  - If the only thing you can think of to name your concrete
    implementation is `FooImpl`, then you *probably* don’t need an
    interface at all, and can just replace the interface with the
    concrete class and give it a good name. (I won’t present this as an
    absolute, but I’ve never yet seen `FooImpl` actually be the right
    choice.)

  - In general, the file name should follow its primary export, so I
    agree that `localInstance.js` is better. The `Impl.js` is just as
    superfluous in the filename; eschew noise.